### PR TITLE
[SPARK-23166] [ML] Add maxDF Parameter to CountVectorizer

### PR DESCRIPTION
**Pull request by ymazari**
## What changes were proposed in this pull request?
Currently, the CountVectorizer has a minDF parameter.

It might be useful to also have a maxDF parameter.
It will be used as a threshold for filtering all the terms that occur very frequently in a text corpus, because they are not very informative or could even be stop-words.

This is analogous to scikit-learn, CountVectorizer, max_df.

## How was this patch tested?

Unit tets.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala [hunk 1]
```diff
@@ -113,7 +132,7 @@ private[feature] trait CountVectorizerParams extends Params with HasInputCol wit
   /** @group getParam */
   def getBinary: Boolean = $(binary)
 
-  setDefault(vocabSize -> (1 << 18), minDF -> 1.0, minTF -> 1.0, binary -> false)
+  setDefault(vocabSize -> (1 << 18), minDF -> 1.0, minTF -> 1.0, binary -> false, maxDF -> Long.MaxValue)
```
Comment (mgaido91) [inline mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala]: what about avoiding to set a default value and apply the filter only if it was set?

File: mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala [hunk 2]
```diff
@@ -169,7 +197,7 @@ class CountVectorizer @Since("1.5.0") (@Since("1.5.0") override val uid: String)
     }.reduceByKey { case ((wc1, df1), (wc2, df2)) =>
       (wc1 + wc2, df1 + df2)
     }.filter { case (word, (wc, df)) =>
-      df >= minDf
+      (df >= minDf) && (df <= maxDf)
```
Comment (mgaido91) [inline mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala]: nit: the parenthesis are not needed

File: mllib/src/test/scala/org/apache/spark/ml/feature/CountVectorizerSuite.scala [hunk 3]
```diff
@@ -119,6 +119,41 @@ class CountVectorizerSuite extends SparkFunSuite with MLlibTestSparkContext
...
+      .setInputCol("words")
+      .setOutputCol("features")
+      .setMaxDF(0.75)
+      .fit(df)
+    assert(cvModel2.vocabulary === Array("b", "c", "d"))
+
+    cvModel2.transform(df).select("features", "expected").collect().foreach {
+      case Row(features: Vector, expected: Vector) =>
+        assert(features ~== expected absTol 1e-14)
+    }
```
Comment (mgaido91) [inline mllib/src/test/scala/org/apache/spark/ml/feature/CountVectorizerSuite.scala]: may you please also add a UT to check that setting both `maxDF` and `minDF` works as expected?

File: mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala (hunk 2 above)
Comment (ymazari) [reply mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala]: Right. I just added them for clarity.

File: mllib/src/test/scala/org/apache/spark/ml/feature/CountVectorizerSuite.scala (hunk 3 above)
Comment (ymazari) [reply mllib/src/test/scala/org/apache/spark/ml/feature/CountVectorizerSuite.scala]: I will.

**Pull request closed without merging**